share/smack/include/pthread.h

```diff
@@ -0,0 +1,71 @@
+//
```

Comment:
After further reflection, I suspect that there is yet a cleaner solution, which is a variation of the original solution. (Or perhaps it was even the original solution, but I did not understand it correctly.)

Monty says: “We are preempting the system's pthread.h file by passing the path to our version of pthread.h on the clang command line,” and that “include guards work, but only if pthread.h is included in the user's program before any other header that recursively includes the pthread typedefs.”

First of all, I don’t see in this pull request where the “[preemption of] the system’s pthread.h” is happening. Where is that actually happening?

However, in principle, shouldn’t it be possible to simply ensure that:

1. our pthread API headers appear in the include paths of all compilation commands, before default system headers, and

2. our pthread API implementation module is linked in?

I would expect the first to appear as a modification to our standard clang compilation command, with some additional flags to specify the path to our headers. If this were done, why would we have to worry about recursive includes and the possibility that default system pthread API headers might be pulled in? Clang’s include path would ensure that our pthread API headers are pulled in, wouldn’t it? Or is it not possible to preempt default system headers in this way?

After all, I think this method would be superior to the type-incompatible linking solution, if you can get rid of the caveat of include frailty.